src/Shared/AssemblyNameExtension.cs

```diff
@@ -121,6 +122,55 @@ internal AssemblyNameExtension(string assemblyName, bool validate) : this()
             }
         }
 
+        /// <summary>
+        /// Ctor for deserializing from state file (binary serialization).
+        /// <remarks>This is required because AssemblyName is not Serializable on .NET Core.</remarks>
+        /// </summary>
+        /// <param name="info"></param>
+        /// <param name="context"></param>
+        private AssemblyNameExtension(SerializationInfo info, StreamingContext context)
+        {
+            var hasAsAssemblyName = info.GetBoolean("asAssemblyName");
+            if (hasAsAssemblyName)
+            {
+                var name = info.GetString("asAssemblyName.Name");
```

Comment:
hopefully there are unit tests that would fail if there was a mistake in the serializatoin/deserialization? it's very easy to have a tpyo or suchlike.